doc/architecture/platforms/index.rst

```diff
@@ -0,0 +1,15 @@
+Platforms
+===================
+
+CCF can run on several platforms, which platform CCF runs on will have impacts on how attestations are verified and therefore what security guarantees you have.
+
+.. toctree::
+    :maxdepth: 1
+
+    sgx
+    snp
+    virtual
+
+Which platform code is run is determined partly by the compile time flag ``COMPILE_TARGETS`` and partly by runtime detection of the hardware.
```

Comment:
`COMPILE_TARGETS` is a contributor concern (i.e. who builds CCF) rather than an operator concern. While we could add a note on this [here](https://microsoft.github.io/CCF/main/contribute/build_ccf.html#build-switches), I think we should omit this from the documentation, until #4069 has been addressed.

Reply:
Part of the aim of this is to have one reference for everything about the platform though so I think it makes sense to include it? We could drop it for now, but I don't see the harm in documenting how it is now and then updating the docs when the implementation changes?

Reply:
Because the docs is separate by area of concern, it's not expected that operators (where this section belongs) compile CCF, or the CCF app so I think it'd be better to point to [`enclave.type`](https://microsoft.github.io/CCF/main/operations/configuration.html#enclave) in the node configuration instead.

Reply:
https://github.com/microsoft/CCF/pull/4330/commits/1263baedcfab78e5488ff8d47c5da46581eb5508